### PR TITLE
.format() coverted to f-strings in segmentation subpackage

### PR DESCRIPTION
**Pull request by Xyno18**
## Description

This PR replaces `.format()` calls on strings within the `segmentation` package with f-strings to partially address issue #5474.

Thank you!

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/segmentation/random_walker_segmentation.py [hunk 1]
```diff
@@ -428,8 +428,8 @@ def random_walker(data, labels, beta=130, mode='cg_j', tol=1.e-3, copy=True,
     # Parse input data
     if mode not in ('cg_mg', 'cg', 'bf', 'cg_j', None):
         raise ValueError(
-            "{mode} is not a valid mode. Valid modes are 'cg_mg',"
-            " 'cg', 'cg_j', 'bf' and None".format(mode=mode))
+            f"{mode} is not a valid mode. Valid modes are 'cg_mg',"
+            " 'cg', 'cg_j', 'bf' and None")
```
Comment (mkcor) [inline skimage/segmentation/random_walker_segmentation.py:432]:
```suggestion
            f"{mode} is not a valid mode. Valid modes are 'cg_mg', "
            f"'cg', 'cg_j', 'bf', and None")
```
out of consistency and as per this discussion https://github.com/scikit-image/scikit-image/pull/5478#discussion_r671584470

**Review by mkcor — COMMENTED** [review]
Thank you! Please add the f-prefix to each line of a multi-line f-string, just to follow a given style convention (see #5478).

File: skimage/segmentation/_clear_border.py [hunk 2]
```diff
@@ -82,9 +82,9 @@ def clear_border(labels, buffer_size=0, bgval=0, in_place=False, mask=None,
 
     if mask is not None:
         err_msg = ("labels and mask should have the same shape but "
-                   "are {} and {}")
+                   f"are {out.shape} and {mask.shape}")
```
Comment (grlee77) [inline skimage/segmentation/_clear_border.py:85]: We were discussing in https://github.com/scikit-image/scikit-image/pull/5478#pullrequestreview-708813111 whether all lines in a group should start with `f` in cases like this. I don't have a strong feeling either way, but we should do the same in this PR. Since we are still discussing, I would make the change yet.

**Review by grlee77 — APPROVED** [review]
Thanks @Xyno18, this looks good to me aside from a comment about style preference

File: skimage/segmentation/_clear_border.py (hunk 2 above)
Comment (grlee77) [reply skimage/segmentation/_clear_border.py:85]: It also came up in that PR to potentially change to single quoted strings while we are already modifying these lines.

**Comment by mkcor** [comment]
@Xyno18 no big deal, but please write commit messages at the imperative :slightly_smiling_face: 

Reference: https://chris.beams.io/posts/git-commit/#imperative

**Comment by Xyno18** [comment]
Any other changes that I should make?

**Comment by mkcor** [comment]
> Any other changes that I should make?

@Xyno18 apparently, you +1'd Greg's comment: "It also came up in that PR to potentially change to single quoted strings while we are already modifying these lines."

I thought this meant you were going to take care of it: Am I mistaken? Only where possible, of course (i.e., not in `skimage/segmentation/random_walker_segmentation.py`).

**Comment by Xyno18** [comment]
@mkcor Apparently I +1's the wrong comment. I meant to +1 this

> We were discussing in #5478 (review) whether all lines in a group should start with f in cases like this. I don't have a strong feeling either way, but we should do the same in this PR. Since we are still discussing, I would make the change yet.

Also, Greg mentioned that it's still under consideration so I didn't make those changes from double-quote to single quote

**Comment by Xyno18** [comment]
Also, I think that it is also possible to make changes in `skimage/segmentation/random_walker_segmentation.py` using escape sequence

**Comment by mkcor** [comment]
> @mkcor Apparently I +1's the wrong comment. I meant to +1 this
> 
> > We were discussing in #5478 (review) whether all lines in a group should start with f in cases like this. I don't have a strong feeling either way, but we should do the same in this PR. Since we are still discussing, I would make the change yet.
> 
> Also, Greg mentioned that it's still under consideration so I didn't make those changes from double-quote to single quote

Ok. Well, in this sense, *both* the multi-line f-prefix *and* the single-quote default are "still under consideration." Personally, I'm a supporter of both for the reasons I mentioned in the #5478 conversation. It would be good to have other @scikit-image/core devs weigh in.

PS: Wasn't "would" in @grlee77's comment above supposed to be "wouldn't?" :thinking:

**Comment by mkcor** [comment]
> Also, I think that it is also possible to make changes in `skimage/segmentation/random_walker_segmentation.py` using escape sequence

@Xyno18 no, this is not recommended: " When a string contains single or double quote characters, however, use the other one to avoid backslashes in the string. It improves readability." (from https://www.python.org/dev/peps/pep-0008/#string-quotes)

File: skimage/segmentation/random_walker_segmentation.py [hunk 3]
```diff
@@ -428,8 +428,8 @@ def random_walker(data, labels, beta=130, mode='cg_j', tol=1.e-3, copy=True,
     # Parse input data
     if mode not in ('cg_mg', 'cg', 'bf', 'cg_j', None):
         raise ValueError(
-            f"{mode} is not a valid mode. Valid modes are 'cg_mg',"
-            f" 'cg', 'cg_j', 'bf', and None")
+            f'{mode} is not a valid mode. Valid modes are \'cg_mg\','
+            f' \'cg\', \'cg_j\', \'bf\', and None')
```
Comment (mkcor) [inline skimage/segmentation/random_walker_segmentation.py:432]: Please revert this change as per my latest [comment](https://github.com/scikit-image/scikit-image/pull/5479#issuecomment-882111317).

**Review by jni — APPROVED** [review]

**Comment by jni** [comment]
Thank you @Xyno18 and welcome to the scikit-image contributors list! 😃 And thanks @grlee77 and @mkcor for the reviews!
